pages/k8s/snaps.md

```diff
@@ -0,0 +1,189 @@
+
+`kubefed` - for running commands on federated clusters. [Docs][kubefed] &#124; [Snap][kubefed-snap]
+
+These are all installed in the same way. For example:
+
+```bash
+sudo snap install kubectl --classic
+```
+
+## Server Snaps
+
+When installing **Charmed Kubernetes**, various necessary services are actually
+installed from snap packages. The installation and management of these snaps is
+undertaken by the charms; there is no need for a user to interact with the
+snaps directly. The details here are provided for information only.
+
+
+| Snap | Type | Store page |
+| kube-apiserver | strict | <https://snapcraft.io/kube-apiserver> |
+| kube-controller-manager | strict  | <https://snapcraft.io/kube-controller-manager>  |
+| kube-proxy | classic | <https://snapcraft.io/kube-proxy> |
+| kube-scheduler  |  strict  | <https://snapcraft.io/kube-scheduler>  |
+| kubelet  | classic  | <https://snapcraft.io/kubelet>  |
+
+### Example: kube-apiserver
+
+We will use kube-apiserver as an example. The other services generally work the
+same way.
+
+Install with `snap install`:
+
+```
+sudo snap install kube-apiserver
+```
+
+This creates a systemd service named `snap.kube-apiserver.daemon`. Initially,
+it will be in an error state because it's missing important configuration:
+
+Running:
+
+```bash
+systemctl status snap.kube-apiserver.daemon
+```
+
+... will return status similar to:
+
+```no-highlight
+● snap.kube-apiserver.daemon.service - Service for snap application kube-apiserver.daemon
+   Loaded: loaded (/etc/systemd/system/snap.kube-apiserver.daemon.service; enabled; vendor preset: enabled)
+   Active: inactive (dead) (Result: exit-code) since Fri 2020-09-01 15:54:39 UTC; 11s ago
+   ...
+```
+
+The snap can be configured using `snap set`:
+
+```bash
+sudo snap set kube-apiserver \
+  etcd-servers=https://172.31.9.254:2379 \
+  etcd-certfile=/root/certs/client.crt \
+  etcd-keyfile=/root/certs/client.key \
+  etcd-cafile=/root/certs/ca.crt \
+  service-cluster-ip-range=10.123.123.0/24 \
+  cert-dir=/root/certs
+```
+
+<div class="p-notification--information">
+  <p class="p-notification__response">
+  Note: Any files used by the service, such as certificate files, must be
+  placed within the /root/ directory to be visible to the service. This
+  limitation allows us to run a few of the services in a strict confinement
+  mode that offers better isolation and security.
+  </p>
+</div>
+
+
+After configuring, restart the service and you should see it running:
+
+```bash
+sudo service snap.kube-apiserver.daemon restart
+systemctl status snap.kube-apiserver.daemon
+```
+```no-highlight
+● snap.kube-apiserver.daemon.service - Service for snap application kube-apiserver.daemon
+   Loaded: loaded (/etc/systemd/system/snap.kube-apiserver.daemon.service; enabled; vendor preset: enabled)
+   Active: active (running) since Fri 2017-09-01 16:02:33 UTC; 6s ago
+   ...
+```
+
+### Configuration
```

Comment:
I think this whole Configuration section can go away. All the content in here is irrelevant when using the newer "args" config, and that config is pretty self-explanatory from the example that comes before this.